### PR TITLE
feat(SlashDrive): enable blindly downloading an encrypted drive

### PR DESCRIPTION
**Pull request by Nazeh**
move the content key to the Hypercore's header, and make sure it is
written unencrypted, regardless of the encryption of the rest of the
metadata btree

BREAKING CHANGE: any previously written slashdrives can't be read by this version or read data created by this version

### CODE REVIEW

**Review by instabot42 — COMMENTED** [review]
What are the security implications of including this data in the header unencrypted. I'm deep enough into the hyperdrive stuff to really understand the full implications.

File: packages/drive/src/index.js [hunk 1]
```diff
@@ -247,11 +223,24 @@ export class SlashDrive extends EventEmitter {
...
   }
 
-  async _onMetadataCoreAppend () {
+  /**
+   * @returns
+   */
+  async download () {
+    if (!this.content) await this.update()
+    const downloadedFeed = this.feed.download()
+    // process.title === 'node' && (await this.content?.core.update());
```
Comment (instabot42) [inline packages/drive/src/index.js:232]: Should this dead code be removed

**Review by instabot42 — CHANGES_REQUESTED** [review]
bit trivial, but... :-)

**Comment by Nazeh** [comment]
> What are the security implications of including this data in the header unencrypted. I'm deep enough into the hyperdrive stuff to really understand the full implications.

I don't see much issues really, at the very least everyone knows it _is_ a hypercore, exposing what type of a core is it isn't that much of an issue. 

Chances are, you are blind seeding your drives with a seeder that already knows that you are using Slashtags, so of course your feeds are slashdrives.

**Pull request merged**